frame/plasm-rewards/src/lib.rs

```diff
@@ -0,0 +1,428 @@
+// This value is used by the `on_runtime_upgrade` logic to determine whether we run
+// storage migration logic. This should match directly with the semantic versions of the Rust crate.
+#[derive(Encode, Decode, Clone, Copy, PartialEq, Eq, RuntimeDebug)]
+enum Releases {
+    V1_0_0Ancient,
+    V2_0_0,
+}
+
+impl Default for Releases {
+    fn default() -> Self {
+        Releases::V2_0_0
+    }
+}
+
+/// Information regarding the active era (era in used in session).
+#[derive(Encode, Decode, RuntimeDebug, PartialEq, Eq)]
+pub struct ActiveEraInfo<Moment> {
+    /// Index of era.
+    index: EraIndex,
+    /// Moment of start
+    ///
+    /// Start can be none if start hasn't been set for the era yet,
+    /// Start is set on the first on_finalize of the era to guarantee usage of `Time`.
+    start: Option<Moment>,
+}
+
+pub trait Trait: session::Trait {
+    /// The staking balance.
+    type Currency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>;
+
+    /// Time used for computing era duration.
+    type Time: Time;
+
+    /// Number of sessions per era.
+    type SessionsPerEra: Get<SessionIndex>;
+
+    /// Number of eras that staked funds must remain bonded for.
+    type BondingDuration: Get<EraIndex>;
+
+    /// Get the amount of staking for dapps per era.
+    type GetForDappsStaking: traits::GetEraStakingAmount<EraIndex, BalanceOf<Self>>;
+
+    /// Get the amount of staking for security per era.
+    type GetForSecurityStaking: traits::GetEraStakingAmount<EraIndex, BalanceOf<Self>>;
+
+    /// How to compute total issue PLM for rewards.
+    type ComputeTotalPayout: traits::ComputeTotalPayout;
+
+    /// Maybe next validators.
+    type MaybeValidators: traits::MaybeValidators<EraIndex, Self::AccountId>;
+
+    /// The overarching event type.
+    type Event: From<Event<Self>> + Into<<Self as system::Trait>::Event>;
+}
+
+decl_storage! {
+    trait Store for Module<T: Trait> as PlasmStaking {
+        /// This is the compensation paid for the dapps operator of the Plasm Network.
+        /// This is stored on a per-era basis.
+        pub ForDappsEraReward get(fn for_dapps_era_reward): map hasher(twox_64_concat) EraIndex => Option<BalanceOf<T>>;
+
+        /// This is the compensation paid for the security of the Plasm Network.
+        /// This is stored on a per-era basis.
+        pub ForSecurityEraReward get(fn for_security_era_reward): map hasher(twox_64_concat) EraIndex => Option<BalanceOf<T>>;
+
+        /// Number of era to keep in history.
+        ///
+        /// Information is kept for eras in `[current_era - history_depth; current_era]`
+        ///
+        /// Must be more than the number of era delayed by session otherwise.
+        /// i.e. active era must always be in history.
+        /// i.e. `active_era > current_era - history_depth` must be guaranteed.
+        pub HistoryDepth get(fn history_depth) config(): u32 = 84;
```

Comment:
Magic number `84`.

Reply:
I refer to:
https://github.com/paritytech/substrate/blob/master/frame/staking/src/lib.rs#L828

I ask the reason to the parity team.

Reply:
The answer:
> 84 eras is roughly 21 days on Kusama - I'm not sure how the value was selected, but that seems like a reasonable length of time for users to claim a payout

Reply:
I added comments that.

Reply:
I mean is better to create constant with a docstring and put the constant name here.